src/store/editor/actionCreators.ts

```diff
@@ -12,6 +12,15 @@ export function updateProjectType(projectType: ProjectType): EditorActionTypes {
     };
 }
 
+export function updateProjectName(projectName: string): EditorActionTypes {
```

Comment:
już nie pamiętam gdzie (czy ActionCreatory powinny to mieć) ale powinieneś wykonać validacje/sanitizację inputu... 
XML to akurat wrażliwy pewnie jest na znaki typu: "<" ">" "/" (ale pewnie i inne również próby) więc musisz poczytać o sanitizowaniu tego gówna

Reply:
hmm. raczej w widoku powinieneś to zrobić. czyli w handlerach na onChange albo coś takiego